src/app/core/interceptors/loader.interceptor.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { LoaderService } from '../services/loader.service';
+import { finalize } from 'rxjs/operators';
+
+@Injectable()
+export class LoaderInterceptor implements HttpInterceptor {
+
+  constructor(public loaderService: LoaderService) { }
+
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
```

Comment:
Podstawowa wersja - nie zadziała jak będą dwa calle w jednym czasie. Lepszym podejściem jest liczenie requestów.
Czyli wchodzi request to counter ++. Na finalize counter --. Loader pokazujemy gdy counter > 0